src/index.ts

```diff
@@ -89,30 +90,56 @@ async function parseArgs() {
       type: 'number',
       default: 1.0,
     })
+    .option('deploy-env', {
+      description: 'Deploy environment',
+      type: 'string',
+      choices: Object.values(DeployEnv),
+      demandOption: true,
+    })
     .epilogue(
       'Always specify arguments as environment variables. Not all arguments are supported as CLI ones yet.',
     ).argv
 
-  return argv
+  const deployEnv = argv['deploy-env'] as DeployEnv
```

Comment:
```suggestion
  const deployEnv = argv['deploy-env']
```
you don't need to type cast if you get rid of the `type: 'string'` part

Reply:
yargs is smarter than you think!

Reply:
separately, it looks like there is a mix of yargs function-scoped config parsing and vanilla module-scoped config parsing going on in this repo. from a quick look, it appears that before this change adding deployEnv to the function-scoped config parsing, there at least wasn't any overlap between the config params that were loaded in the 2 different places/ways. 

I find the overlap unsettling.. reminds me of a bug I've seen where the same env var was defined in some unencrypted config file and a secret that was loaded into memory and over-wrote it, except in this case, the bug would be even more devious, because the same config param could be different in the 2 different places it's loaded. (Here, secrets are loaded before reading from process.env, but in config.ts, I think it's the opposite.)

You can punt on this if you want, but I think there should at least be a Linear issue for following up and unifying the way configs are done in blockchain-api. And maybe also a code comment TODO that links to it

Reply:
Yeah, I was also a bit concerned by the two separate config parsing approaches this repo takes; one locally scoped and one global. I think I'm going to punt for this PR since it's already pretty huge, but will spin up a Linear issue to unify the config approaches.